config/v1/types_descheduling.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Descheduler holds cluster-wide config information to run the Kubernetes Descheduler
+// and influence its placement decisions. The canonical name for this config is `cluster`.
+type Descheduler struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec DeschedulerSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status DeschedulerStatus `json:"status"`
+}
+
+type DeschedulerSpec struct {
+	// Policy contains the name of an upstream Descheduler policy configmap in the openshift-kube-descheduler-operator project
+	Policy ConfigMapNameReference `json:"polocy"`
+	// DeschedulingIntervalSeconds is the number of seconds between descheduler runs
+	// +optional
+	DeschedulingIntervalSeconds *int32 `json:"deschedulingIntervalSeconds,omitempty"`
+	// Flags for descheduler.
+	// +optional
+	Flags []string `json:"flags,omitempty"`
```

Comment:
Looking at the descheduler flags (https://github.com/kubernetes-sigs/descheduler/blob/master/cmd/descheduler/app/options/options.go) I don't think this is needed.

Reply:
True, though I'm replacing it with a `DryRun bool` since that's the only flag we don't have other config options for. (it's available in the upstream descheduler componentconfig api but that isn't used that I'm aware of). should make a TODO for that upstream

Reply:
I still argue about the value of dry run. Do you have valuable use-case for that?

Reply:
It shows all the pods that would be evicted, so it's helpful for a user to know the scope of any changes (like turning on a new policy) before actually starting evictions in a cluster